File: back-node/src/entity/quizChoice/QuizChoice.ts
import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { Quiz } from '../quiz/Quiz';

@Entity()
export class QuizChoice {
  @PrimaryGeneratedColumn()
  choiceId: number | undefined;

  @Column()
  choiceContent: string;

  @Column()
  isAnswer: boolean;

  @ManyToOne(() => Quiz, { eager: false })
  @JoinColumn({ name: 'quiz_id' })
  quiz: Quiz;

  constructor(choiceContent: string, isAnswer: boolean, quiz: Quiz, choiceId?: number) {
    this.choiceId = choiceId;
    this.choiceContent = choiceContent;
    this.isAnswer = isAnswer;
    this.quiz = quiz;
  }
}

File: back-spirng/jobseeckerstudio/quiz/src/main/java/com.jobseeckerstudio.quiz/entity/quizChoice/QuizChoice.java
package com.jobseeckerstudio.user.quiz.entity.quizChoice;

import com.jobseeckerstudio.user.quiz.entity.quiz.Quiz;

import javax.persistence.*;


@Entity
public class QuizChoice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long choiceId;

    @Column(nullable = false)
    private String choiceContent;

    @Column(nullable = false)
    private boolean isAnswer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="quizId")
    private Quiz quiz;
}
